File: pyq/digit_sum.py
# Program to find sum of digits in an alphanumeric string

# Get input string
string = input("Enter an alphanumeric string: ")

# Initialize sum
digit_sum = 0

# Iterate through each character
for char in string:
    # Check if character is a digit
    if char.isdigit():
        # Convert to integer and add to sum
        digit_sum += int(char)

# Print result
print(f"\nInput string: {string}")
print(f"Sum of digits: {digit_sum}") 

File: pyq/digit_sum.c
#include <stdio.h>
#include <string.h>

int main() {
    char str[100];
    int sum = 0;
    
    // Get input string
    printf("Enter an alphanumeric string: ");
    gets(str);
    
    // Iterate through each character
    for(int i = 0; str[i] != '\0'; i++) {
        // Check if character is a digit
        if(str[i] >= '0' && str[i] <= '9') {
            // Convert to integer and add to sum
            sum += (str[i] - '0');
        }
    }
    
    // Print result
    printf("\nInput string: %s\n", str);
    printf("Sum of digits: %d\n", sum);
    
    return 0;
} 